dagster_ray/kuberay/pipes.py

```diff
@@ -112,10 +119,33 @@ def run(  # type: ignore
             ) as job_submission_client:
                 self._job_submission_client = job_submission_client
 
+                session.report_launched(
+                    {
+                        "extras": {
+                            PIPES_LAUNCHED_EXTRAS_RAY_JOB_ID_KEY: ray_job_id,
+                            PIPES_LAUNCHED_EXTRAS_RAY_ADDRESS_KEY: job_submission_client.get_address(),
+                        }
+                    }
+                )
+
                 try:
-                    self._read_messages(context, start_response)
                     self._wait_for_completion(context, start_response)
-                    return PipesClientCompletedInvocation(session)
+
+                    if isinstance(self._message_reader, PipesRayJobMessageReader) and self.port_forward:
```

Comment:
Actually there should just be a separate self-sufficient `PipesKubeRayJobMessageReader` that would manage it's own port-forwarding session.